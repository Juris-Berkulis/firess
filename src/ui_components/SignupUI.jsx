import React from 'react';
import { Link } from 'react-router-dom';

export const SignupUI = (props) => {
    return (
    <div>
        <form onSubmit={props.handleSubmit}>
        <p>Заполните форму ниже, чтобы зарегистрировать новую учетную запись.</p>
        <div>
            <input
            placeholder="Email"
            name="email"
            type="email"
            onChange={props.handleEmailChange}
            value={props.email}
            data-testid="idEmail"
            />
        </div>
        <div>
            <input
            placeholder="Password"
            name="password"
            type="password"
            onChange={props.handlePassChange}
            value={props.password}
            data-testid="idPassword"
            />
        </div>
        <div>
            {props.error && <p data-testid="idError">{props.error}</p>}
            <button type="submit" data-testid="idBtnSubmit">Зарегистрироваться</button>
        </div>
        <hr />
        <p>
            <span>Уже есть аккаунт? </span>
            <Link to={props.allAppComponentsWithPageTitle.login.path}>Войти</Link>
        </p>
        </form>
    </div>
    )
};
