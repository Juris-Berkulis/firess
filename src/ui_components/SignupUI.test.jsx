import { render, screen, fireEvent, act } from "@testing-library/react"
import { SignupUI } from "./SignupUI.jsx";
import '@testing-library/jest-dom';
import { BrowserRouter } from "react-router-dom";

const handleSubmit = jest.fn();
const handleEmailChange = jest.fn();
const handlePassChange = jest.fn();

describe('SignUpUI component', () => {
    it('SignUpUI snapshot', () => {
        const component = render(<BrowserRouter><SignupUI 
            handleSubmit={handleSubmit} 
            handleEmailChange={handleEmailChange} 
            handlePassChange={handlePassChange}
            error='' 
            email='example@example.com' 
            password='12345678'
        /></BrowserRouter>);

        component.debug();
        expect(component).toMatchSnapshot();
    });

    it('SignUpUI renders with a <button> and a "зарегистрироваться" text', () => {
        const component = render(<BrowserRouter><SignupUI 
            handleSubmit={handleSubmit} 
            handleEmailChange={handleEmailChange} 
            handlePassChange={handlePassChange}
            error='' 
            email='example@example.com' 
            password='12345678'
        /></BrowserRouter>);

        expect(component.getByText(/зарегистрироваться/i)).toBeInTheDocument();
        expect(component.getByRole('button')).toBeInTheDocument();
    });

    it('Was the function called?', () => {
        const component = render(<BrowserRouter><SignupUI 
            handleSubmit={handleSubmit} 
            handleEmailChange={handleEmailChange} 
            handlePassChange={handlePassChange}
            error='' 
            email='example@example.com' 
            password='12345678'
        /></BrowserRouter>);

        const submitBtn = component.queryByTestId('idBtnSubmit');

        act(() => {
            fireEvent.click(submitBtn);
        });

        expect(handleSubmit).toBeCalled();
    });

    it('Text input fields have the text', () => {
        const component = render(<BrowserRouter><SignupUI 
            handleSubmit={handleSubmit} 
            handleEmailChange={handleEmailChange} 
            handlePassChange={handlePassChange}
            error='' 
            email='example@example.com' 
            password='12345678'
        /></BrowserRouter>);

        const fieldEmail = component.queryByTestId('idEmail');
        const fieldPassword = component.queryByTestId('idPassword');

        expect(screen.queryByDisplayValue('example@example.com')).toBe(fieldEmail);
        expect(screen.queryByDisplayValue('12345678')).toBe(fieldPassword);
    });

    it('Error renders if it is', () => {
        const error = 'The email address is already in use by another account.';

        const component = render(<BrowserRouter><SignupUI 
            handleSubmit={handleSubmit} 
            handleEmailChange={handleEmailChange} 
            handlePassChange={handlePassChange}
            error={error} 
            email='example@example.com' 
            password='12345678'
        /></BrowserRouter>);

        expect(component.getByText(error)).toBeInTheDocument();

        expect(component.getByTestId('idError'));
    });

    it("Error don't renders if it isn't", () => {
        const component = render(<BrowserRouter><SignupUI 
            handleSubmit={handleSubmit} 
            handleEmailChange={handleEmailChange} 
            handlePassChange={handlePassChange}
            error=''
            email='example@example.com' 
            password='12345678'
        /></BrowserRouter>);

        const errorItem = component.queryByTestId('idError');
        expect(errorItem).toBeNull();
    });

    it('functions call if user is writing in text input fields', () => {
        const component = render(<BrowserRouter><SignupUI 
            handleSubmit={handleSubmit} 
            handleEmailChange={handleEmailChange} 
            handlePassChange={handlePassChange}
            error='' 
            email='' 
            password=''
        /></BrowserRouter>);

        const fieldEmail = component.queryByTestId('idEmail');
        const fieldPassword = component.queryByTestId('idPassword');

        act(() => {
            fireEvent.change(fieldEmail, {
                target: {
                    value: 'login',
                }
            });
            fireEvent.change(fieldPassword, {
                target: {
                    value: 'password',
                }
            });
        });

        expect(handleEmailChange).toBeCalled();
        expect(handlePassChange).toBeCalled();
    });
});
