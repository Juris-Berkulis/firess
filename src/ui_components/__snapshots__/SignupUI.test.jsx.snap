// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpUI component SignUpUI snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form>
          <p>
            Заполните форму ниже, чтобы зарегистрировать новую учетную запись.
          </p>
          <div>
            <input
              data-testid="idEmail"
              name="email"
              placeholder="Email"
              type="email"
              value="example@example.com"
            />
          </div>
          <div>
            <input
              data-testid="idPassword"
              name="password"
              placeholder="Password"
              type="password"
              value="12345678"
            />
          </div>
          <div>
            
            <button
              data-testid="idBtnSubmit"
              type="submit"
            >
              Зарегистрироваться
            </button>
          </div>
          <hr />
          <p>
            <span>
              Уже есть аккаунт? 
            </span>
            <a
              href="/login"
            >
              Войти
            </a>
          </p>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form>
        <p>
          Заполните форму ниже, чтобы зарегистрировать новую учетную запись.
        </p>
        <div>
          <input
            data-testid="idEmail"
            name="email"
            placeholder="Email"
            type="email"
            value="example@example.com"
          />
        </div>
        <div>
          <input
            data-testid="idPassword"
            name="password"
            placeholder="Password"
            type="password"
            value="12345678"
          />
        </div>
        <div>
          
          <button
            data-testid="idBtnSubmit"
            type="submit"
          >
            Зарегистрироваться
          </button>
        </div>
        <hr />
        <p>
          <span>
            Уже есть аккаунт? 
          </span>
          <a
            href="/login"
          >
            Войти
          </a>
        </p>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
